app/bundles/EmailBundle/Helper/MailHelper.php

```diff
@@ -1423,9 +1425,15 @@ public function parsePlainText($content = null): void
             }
         }
 
-        $request = $this->factory->getRequest();
+        $request = $this->requestStack->getMainRequest();
+        if (null !== $request) {
+            $baseUrl = $request->getSchemeAndHttpHost().$request->getBasePath();
+        } else {
+            $baseUrl = $this->factory->getParameter('site_url'); // iKnow this gets the parameter from deprecated service.
```

Comment:
I see the comment, but why not use the CoreParametersHelper here instead? It is already available.